bin/apisix

```diff
@@ -609,6 +632,22 @@ local function init()
         sys_conf["worker_processes"] = "auto"
     end
 
+    if sys_conf["enable_local_dns"] == true then
+        local dns_addrs, err = local_dns_resolver("/etc/resolv.conf")
+        if not dns_addrs then
+            error("failed to import local DNS: " .. err)
+        end
+
+        local dns_resolver = sys_conf["dns_resolver"]
+        if dns_resolver then
+            for _, addr  in ipairs(dns_addrs) do
+                table.insert(dns_resolver, addr)
```

Comment:
`dns_resolver` is for what?  not used after this line.

Reply:
this is not a small PR, please add test cases, otherwise I can not merge it.

Reply:
the `dns_resolver` used in `ngx_tpl`,

https://github.com/apache/incubator-apisix/pull/1217/files/ceee77794539d2b3dee3e90d41096aed2d976b51#diff-62edadffc237f13dc28a694080d293a8R120

https://github.com/apache/incubator-apisix/pull/1217/files/ceee77794539d2b3dee3e90d41096aed2d976b51#diff-62edadffc237f13dc28a694080d293a8R249